test/src/test/java/jenkins/security/RekeySecretAdminMonitorTest.java

```diff
@@ -112,58 +128,67 @@ public void _testBasicWorkflow() throws Exception {
         assertEquals(0, DomNodeUtil.selectNodes(manage, "//*[class='warning']").size());
 
         // and the data should be rewritten
-        verifyRewrite(jenkins.getRootDir());
+        verifyRewrite(j.jenkins.getRootDir());
         assertTrue(monitor.isDone());
 
         // dismiss and the message will be gone
         assertTrue(monitor.isEnabled());
         form = getRekeyForm(wc);
-        submit(form, "dismiss");
+        submit(wc, form, "dismiss");
         assertFalse(monitor.isEnabled());
-        try {
-            getRekeyForm(wc);
-            fail();
-        } catch (ElementNotFoundException e) {
-            // expected
-        }
+        assertThrows(ElementNotFoundException.class, () -> getRekeyForm(wc));
     }
 
-    private HtmlForm getRekeyForm(WebClient wc) throws IOException, SAXException {
+    private HtmlForm getRekeyForm(JenkinsRule.WebClient wc) throws IOException, SAXException {
         return wc.goTo("manage").getFormByName("rekey");
     }
 
+    private void submit(JenkinsRule.WebClient wc, HtmlForm form, String name) throws IOException {
+        WebRequest request = form.getWebRequest(null);
+        /*
+         * TODO There is a long-undiagnosed issue with the test harness not being simpatico with
+         * message.groovy's f.submit. Work around this by matching the behavior of a real browser
```

Comment:
Interesting. Is this Groovy related, or just `f:submit` in `message` views in general? Do we track this somewhere?

Also,
```suggestion
         * TODO There is a long-undiagnosed issue with the test harness not being compatible with
         * message.groovy's f.submit. Work around this by matching the behavior of a real browser
```

Reply:
Not sure. I think commit f6662a8 comes closest to describing the problem.